mlx/ops.cpp

```diff
@@ -2897,4 +2897,27 @@ array tensordot(
   return reshape(matmul(x, y, s), rshape, s);
 }
 
+array outer(const array& a, const array& b, StreamOrDevice s /** = {}*/) {
+  if (a.ndim() != 1 || b.ndim() != 1) {
+    throw std::invalid_argument("[outer] a and b must be 1-dimensional.");
+  }
```

Comment:
Let's follow numpy and flatten if not 1d? https://numpy.org/doc/stable/reference/generated/numpy.outer.html